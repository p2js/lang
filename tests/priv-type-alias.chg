type Point<T> constructible(mkPoint) = { x: T, y: T };

function mkPoint<T>(x: T, y: T): Point<T> {
    return .{ x: x, y: y };
}

function notAllowed(): Point<i32> {
    return .{ x: 1, y: 2 };
}

function main(): i32 {
    return mkPoint(1, 2).x + notAllowed().y;
}
