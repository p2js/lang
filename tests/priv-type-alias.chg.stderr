
    throw new Error('Todo: ' + what);
    ^

Error: Todo: Alias












